refactor: single code base for both text and drawables & other bug fixes

adityastic opened the pull request:
Fixes #189 

Changes: 
- Optimized Code
- Concept of Text is completely removed, we convert everything to the hexadecimal strings and then convert it to byte array
- Renamed BitmapDatatoSend to DataToSend because it is a single class now.
- Fixed a Bug which did not let you send the drawable to device because of the `content.text.isEmpty()`. Now it checks the radio button and then sends the bytearray
- Removed unused resources from the app
- Removed Kotlin Unit Tests as they depended on the Text Logic
- Lint Errors Fixed

adityastic commented:
@iamareebjamal fixed almost all lint errors, excluding the missing icon. Please have a look

iamareebjamal commented:
Peer Review First

adityastic commented:
> Peer Review First

On it Sir.@yashk2000 @yatri1609 @anhanh11001 Please Peer Review

yashk2000 commented on app/src/main/res/values/strings.xml:
```diff
@@ -20,14 +20,14 @@
     <string name="send_text">Send Text</string>
     <string name="send_drawables">Send Drawables</string>
     <string name="invertLED">Invert LED</string>
+    <string name="radio_error">Select option before proceeding</string>
```
Where is this used?

adityastic replied:
Ah I removed the use. Thanks sir, just a min

adityastic replied:
Have loik

adityastic replied:
Fixed, have a look sir

adityastic commented:
@yashk2000 Thanks. @iamareebjamal Peer review done

iamareebjamal commented on app/build.gradle:
```diff
@@ -38,6 +38,8 @@ spotless {
 
 dependencies {
+    // Kotlin Libraries
+    implementation "org.jetbrains.kotlin:kotlin-reflect:$kotlin_version"
```
Why is this needed. It is 2 MB

iamareebjamal replied:
First of all, I see no use of this library, and even if it was used, I would have voted against it

adityastic replied:
Alright sir. I added this because lint told me too

adityastic replied:
Let me remove it

iamareebjamal commented on app/src/main/AndroidManifest.xml:
```diff
@@ -14,13 +14,16 @@
     <application
         android:name=".ui.App"
         android:allowBackup="true"
+        android:fullBackupContent="true"
```
Why

adityastic replied:
using fullBackupContent backs up the user data when a person uninstalls the application, he can restore the state after reinstalling the application

iamareebjamal replied:
How is it related to this issue?

adityastic replied:
It was one of the lint fixes suggested @iamareebjamal.

iamareebjamal commented on app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt:
```diff
@@ -328,13 +310,39 @@ class MessageActivity : AppCompatActivity() {
+                                Mode.values()[mode.selectedItemPosition]
+                        )
+                )
+        )
```
Too much indentation

adityastic replied:
Ok sir, let me make it a bit compact

iamareebjamal commented on app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt:
```diff
@@ -328,13 +310,39 @@ class MessageActivity : AppCompatActivity() {
+                                Mode.values()[mode.selectedItemPosition]
+                        )
+                )
+        )
```
Same

adityastic replied:
On it

iamareebjamal commented on app/src/main/java/com/nilhcem/blenamebadge/util/Converters.kt:
```diff
@@ -137,4 +138,22 @@ object Converters {
+                            invertHex(DataToByteArrayConverter.CHAR_CODES.getValue(letter))
+                        else
+                            DataToByteArrayConverter.CHAR_CODES.getValue(letter)
+                )
```
Too much indentation

adityastic replied:
On it sir

adityastic commented:
@iamareebjamal Thank you sir, I found 8 tab indentation issues in many files. I've changed all of them

adityastic commented:
@iamareebjamal have a look at the fixes sir

iamareebjamal commented on app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt:
```diff
@@ -331,13 +310,39 @@ class MessageActivity : AppCompatActivity() {
-        return DataToSend(listOf(Message(content.text.trim().toString(), flash.isChecked, marquee.isChecked, Speed.values()[speed.selectedItemPosition], Mode.values()[mode.selectedItemPosition])), invertLED.isChecked)
+    private fun convertTextToDeviceDataModel(): DataToSend {
+        return DataToSend(
+            listOf(
```
Move to previous line

adityastic replied:
yes sir, I am learning the type of indentation required now. just a min

adityastic replied:
made the changes, have a look

iamareebjamal commented on app/src/main/java/com/nilhcem/blenamebadge/ui/message/MessageActivity.kt:
```diff
@@ -331,13 +310,39 @@ class MessageActivity : AppCompatActivity() {
+    private fun convertTextToDeviceDataModel(): DataToSend {
+        return DataToSend(
+            listOf(
+                Message(
```
Move to previous line

adityastic replied:
ok sir

naman653 commented on app/src/main/res/values/strings.xml:
```diff
@@ -20,14 +20,13 @@
-    <string name="txt_turn_on_bluetooth">Turn on Bluetooth and try again</string>
     <string name="no_device_found">No device found. Please try again.</string>
     <string name="select_drawable_text">Select Drawable:</string>
-    <string name="select_drawable">Please select Drawable to be previewed</string>
```
Please don't remove these strings for now. Might be useful.

adityastic replied:
Sir, we are using automatic preview now, the second one will definitely not be used. We can keep the first one though

naman653 commented on app/src/main/res/values/strings.xml:
```diff
@@ -20,14 +20,13 @@
     <string name="send_button">SEND</string>
-    <string name="preview_button">PREVIEW</string>
     <string name="permission_required">Permission required</string>
-    <string name="txt_turn_on_bluetooth">Turn on Bluetooth and try again</string>
```
Same

adityastic replied:
Yes Sir, sure. let me undo that

yashk2000 replied:
Is this resource really needed @naman653? We are using `enable_bluetooth` to warn the user when bluetooth is off. This string is redundant in any case according to me.

adityastic replied:
@naman653 Yeah TBH @yashk2000 is right

adityastic commented:
@iamareebjamal @naman653 Done, check the new changes. let me know if anything else is to be fixed

adityastic commented:
@iamareebjamal when I was going through the code right now. I also found that KeyboardUtils created is a duplicate. We already have a view extension class called 
[https://github.com/fossasia/badge-magic-android/blob/development/app/src/main/java/com/nilhcem/blenamebadge/core/android/ext/ViewExt.kt](url)
 that does this for us. So, I removed it in this update and fixed the working logic
